### ISSUE TITLE
Improve runtime of PCSF by unrooting PCST

### ISSUE THREAD

**Issue by iamjli** [fraenkel-lab/OmicsIntegrator2#98]
See [this issue](https://github.com/fraenkel-lab/pcst_fast/issues/7#issuecomment-465389322).

**Comment by zfrenchee** [comment]
@iamjli could you offer more detail on how this would work?

**Comment by iamjli** [comment]
Sure. 

One way to force a node to be root is simply assigning a large prize to it. This would allow us to use PCST in unrooted mode (which is faster), while effectively giving a rooted solution. 

In practice, the value of the prize simply needs to be greater than the value of the edge costs connecting the dummy node (`w`). This works because it should always be worth picking up the dummy node since the prize is greater than the cost on the edges.

**Comment by zfrenchee** [comment]
Good idea. A good way to do this would be to expose the choice of whether to have the [pcst_fast](https://github.com/fraenkel-lab/OmicsIntegrator2/blob/82245952a555c20e34b346009e3d59c956be99d9/src/graph.py#L295) routine be rooted or unrooted in the [pcsf](https://github.com/fraenkel-lab/OmicsIntegrator2/blob/82245952a555c20e34b346009e3d59c956be99d9/src/graph.py#L251) method.

**Comment by zfrenchee** [comment]
Do you want to take this on, @iamjli ?

**Comment by sgosline** [comment]
Hi, just chiming in here out of curiosity - the rooted option was only a bit of a 'hack' to get from solving a tree to solving a forest. If there is a way to get a forest without the dummy node, couldn't we just move toward that?

**Comment by zfrenchee** [comment]
@sgosline thanks for the suggestion. [pcst_fast](https://github.com/fraenkel-lab/pcst_fast) has a num_clusters option, which specifies a hard constraint on the number of trees in the output forest. The dummy node / omega-weighted edges provides for a soft constraint on the number of outcome clusters, which I think is desirable enough to keep around.

**Comment by iamjli** [comment]
Okay, done. 

I did some tests and the run time didn't improve all that much for unrooted PCSF. In addition, the unrooted variant performed worse as measured by the objective value. I definitely see run time improvement in my separate PCSF implementation, I'm guessing it's dependent on the structure of the graph and data. 

So I've kept the default to be "rooted" in this PR, I'll leave it up to @zfrenchee to see if it's worth merging [this PR](https://github.com/fraenkel-lab/OmicsIntegrator2/pull/99).
